fix: webapp screen reader accessibility improvements

arjita-mitra opened the pull request:
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
 - webapp screen reader accessibility improvements

- The **PR Description**
  - overall webapp screen reader accessibility improvements
----

# What's new in this PR?

### Issues

- https://docs.google.com/spreadsheets/d/1g5ENPrk7NivDOJy9vKoymXnDvu42G3rlR9oKzYslWBs/edit?pli=1#gid=0

### Solutions

- adding accessible controls for screen reader

### Testing

#### How to Test

- need to run screen readers for areas mentioned in the doc
----
##### References
1. https://docs.google.com/spreadsheets/d/1g5ENPrk7NivDOJy9vKoymXnDvu42G3rlR9oKzYslWBs/edit?pli=1#gid=0
2. https://wearezeta.atlassian.net/browse/ACC-21

arjita-mitra committed:
feat: improve accessibility(ACC-21)

improve accessibility of conversation info, search people, devices, profile picture screens

arjita-mitra committed:
Merge branch 'dev' of https://github.com/wireapp/wire-webapp into feature/ACC-21

arjita-mitra committed:
feat: improve accessibility of profile, conversation,  devices,  group participants, chat messages, edit profile(ACC-21)

arjita-mitra committed:
Merge branch 'dev' of https://github.com/wireapp/wire-webapp into feature/ACC-21

arjita-mitra committed:
fix: delete for me/everyone modal close btn aria label(ACC-21)

arjita-mitra committed:
fix: accessibility modal close button
accordion accessibility controls(ACC-21)

arjita-mitra committed:
Merge branch 'acc' of https://github.com/wireapp/wire-webapp into feature/ACC-21

arjita-mitra committed:
fix: remove duplicate entry for translation(ACC-21)

przemvs commented on src/script/components/TextInput/TextInput.tsx:
```diff
@@ -111,13 +116,28 @@ const TextInput: React.ForwardRefRenderFunction<HTMLDivElement, UserInputProps>
+          onClick={onCancel}
+          aria-label={t('accessibility.userProfileDeleteEntry')}
+          onKeyDown={event => {
+            if (event.shiftKey && event.key === 'Tab') {
```
Please use isTabKey from KeyboardUtils :)

przemvs commented on src/script/components/TextInput/TextInput.tsx:
```diff
@@ -111,13 +116,28 @@ const TextInput: React.ForwardRefRenderFunction<HTMLDivElement, UserInputProps>
+            if (event.shiftKey && event.key === 'Tab') {
+              // shift+tab from clear button should focus on the input field
+              setIsEditing(true);
+            } else if (event.key === 'Tab') {
```
Please use isTabKey from KeyboardUtils :)

przemvs commented on src/script/page/MainContent/panels/preferences/accountPreferences/AccountInput.tsx:
```diff
@@ -182,20 +184,32 @@ const AccountInput: React.FC<AccountInputProps> = ({
               (event.target as HTMLInputElement).blur();
+              // on enter save changes and close the editable field
+              setIsEditing(false);
+            } else if (event.key === 'Tab' && !!!input) {
```
Please use isTabKey from KeyboardUtils :)

przemvs commented on src/script/page/MainContent/panels/preferences/accountPreferences/AccountInput.tsx:
```diff
@@ -182,20 +184,32 @@ const AccountInput: React.FC<AccountInputProps> = ({
+            setIsEditing(true);
+          }}
           onKeyDown={event => {
             if (event.key === 'Enter' && !event.shiftKey && !event.altKey) {
```
I know is not your code, but please change event.key === 'Enter' to isEnterKey :)

przemvs reviewed (changes requested):
few changes :)

V-Gira reviewed (commented):
I don't understand why testing is failling on GroupCreationModal. It passes locally and CI passed with no issues last PR

atomrc commented:
> I don't understand why testing is failling on GroupCreationModal. It passes locally and CI passed with no issues last PR

@V-Gira there is a typecheck error because the `TextInput` component has changed with required props, that are not passed in the `GroupCreationModal` component :)

atomrc commented on src/script/components/TextInput/TextInput.tsx:
```diff
@@ -29,6 +29,7 @@ import {
   getInputCSS,
   getLabelCSS,
 } from 'Components/TextInput/TextInput.styles';
```
```suggestion
} from './TextInput.styles';
```

atomrc commented on src/script/page/MainContent/panels/preferences/accountPreferences/AccountInput.tsx:
```diff
@@ -99,7 +101,7 @@ const AccountInput: React.FC<AccountInputProps> = ({
       document.removeEventListener('mousedown', handleClickOutside);
     };
-  }, [textInputRef]);
+  }, [inputWrapperRef]);
```
This will never trigger since it's a `ref`, react doesn't listen to `ref` changed

atomrc replied:
well, to be fair, the previous code should not work as well.. 🙃

atomrc commented on src/script/page/MainContent/panels/preferences/accountPreferences/AccountInput.tsx:
```diff
@@ -182,20 +184,32 @@ const AccountInput: React.FC<AccountInputProps> = ({
           label={label}
           name={valueUie ? valueUie : fieldName}
           value={input}
+          inputWrapperRef={inputWrapperRef}
```
This ref is never `set` (passed in a `ref={inputWrapperRef}` prop) it will always be `null` here

atomrc commented on src/script/page/LeftSidebar/panels/Conversations/GroupedConversationsFolder.tsx:
```diff
@@ -55,7 +55,7 @@ const GroupedConversationsFolder: React.FC<GroupedConversationsFolderProps> = ({
     <li className="conversation-folder" data-uie-name="conversation-folder" data-uie-value={folder.name}>
       <GroupedConversationHeader onClick={() => toggle(folder.id)} conversationLabel={folder} isOpen={isExpanded} />
-      <ul css={css({listStyle: 'none', padding: 0})}>
+      <ul css={css({listStyle: 'none', padding: 0})} id={folder.id}>
```
why is this needed?

arjita-mitra replied:
This is for aria-controls. 
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-controls
@atomrc

atomrc replied:
thanks for the clarification 👍

arjita-mitra committed:
Update src/script/components/TextInput/TextInput.tsx

Co-authored-by: Thomas Belin <thomasbelin4@gmail.com>

arjita-mitra committed:
fix: code review comments addressed(ACC-21)

remove input ref forwarding to child component
use isTabKey and isEnterKey util

arjita-mitra committed:
Merge branch 'feature/ACC-21' of https://github.com/wireapp/wire-webapp into feature/ACC-21

arjita-mitra committed:
Merge branch 'acc' of https://github.com/wireapp/wire-webapp into feature/ACC-21

arjita-mitra committed:
fix: make ref and editing function optional(ACC-21)
